src/benchmarklib/tpch/tpch_benchmark_item_runner.cpp

```diff
@@ -27,40 +28,50 @@ std::string calculate_date(boost::gregorian::date date, int months, int days = 0
 
 namespace opossum {
 
-TPCHQueryGenerator::TPCHQueryGenerator(bool use_prepared_statements, float scale_factor)
-    : _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
-  _selected_queries.resize(22);
-  std::iota(_selected_queries.begin(), _selected_queries.end(), QueryID{0});
+TPCHBenchmarkItemRunner::TPCHBenchmarkItemRunner(const std::shared_ptr<BenchmarkConfig>& config,bool use_prepared_statements, float scale_factor)
+    : AbstractBenchmarkItemRunner(config), _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
+  _selected_items.resize(22);
+  std::iota(_selected_items.begin(), _selected_items.end(), BenchmarkItemID{0});
+  _prepare_queries();
 }
 
-TPCHQueryGenerator::TPCHQueryGenerator(bool use_prepared_statements, float scale_factor,
-                                       const std::vector<QueryID>& selected_queries)
-    : _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
-  _selected_queries = selected_queries;
+TPCHBenchmarkItemRunner::TPCHBenchmarkItemRunner(const std::shared_ptr<BenchmarkConfig>& config,bool use_prepared_statements, float scale_factor,
+                                                 const std::vector<BenchmarkItemID>& selected_items)
+    : AbstractBenchmarkItemRunner(config), _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
+  _selected_items = selected_items;
```

Comment:
Why is this not a virtual method call? The way it is now, nothing enforces that the derived *BenchmarkItemRunner actually supplies this information.

Reply:
I can't follow. If you don't supply it, the other constructor will be used.

Reply:
I meant: Why is _selected_items a base class member variable (that a derived class needs to remember to set!!) instead of having `virtual AbstractBenchmarkItemRunner::selected_items() = 0` which forces deriving classes to implement it.

Reply:
Gotcha.